community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -131,7 +131,7 @@ TransactionType upgradeToSchemaTransaction() throws InvalidTransactionTypeKernel
     private TransactionHooks.TransactionHooksState hooksState;
     private KernelStatement currentStatement;
```

Comment:
fully agree with all your statements, but still TransactionType contains `READ` which should not be there anymore because we have `AccessMode` now right?